### ISSUE TITLE
wizard: collect bugs / improvements

### ISSUE THREAD

**Issue by lucasrodes** [owid/etl#1563]
After #1539 you can now use the new tool `wizard` to generate templates for your ETL steps.

Use this issue to report bugs that you may encounter. If the issue is very complex, feel free to create a separate - and more extense - issue.

**Comment by pabloarosado** [comment]
I'm using wizard now for the first time, so I'll write here all issues I find along the way:
* When I open wizard, I get the warning:
```
2023-09-01 08:37:55.275 WARNING streamlit.runtime.caching.cache_data_api: No runtime found, using MemoryCacheStorageManager
['streamlit', 'run', '/Users/prosado/Documents/owid/repos/etl/apps/wizard/app.py', '--server.port', '8053', '--', '--phase', 'all', '--run-checks']
```

**Comment by lucasrodes** [comment]
These issues should be solved in https://github.com/owid/etl/pull/1567

**Comment by lucasrodes** [comment]
@Marigold @pabloarosado From our discussions and the documentation on Notion, I had assumed that we agreed that the field `license` should be under `$.meta.origin`. 

However, from what Pablo is saying, it sounds like this is raising an error. As a solution for now, I've set wizard to export the field `license` both at `$.meta.origin.license` and `$.meta.license` levels.

Do you think this is fine?

**Comment by Marigold** [comment]
> It took me a while to realise that the produced .dvc file had license indented too many spaces, hence becoming an item of of origin. I think they are meant to be separate fields in the snapshot metadata.

The original idea was to keep it **under origin**, not separate from it. `license` is there only for backward compatibility. Ideally, snapshot meta would have either `source & license` or `origin & origin.license` (we could refactor old files if this causes too much confusion).

**Comment by lucasrodes** [comment]
walkthrough bug that might be occurring here too: https://github.com/owid/etl/issues/1571

**Comment by paarriagadap** [comment]
I see that `update_period_days` is mandatory in the guideline, but it is not available by default in any of the YAML files

**Comment by paarriagadap** [comment]
- [ ]

**Comment by paarriagadap** [comment]
Two related comments to the ones above:
- [ ] `description_processing` is not shown by default when walkthrough garden generates the yaml, and it's a recommended field
- [ ] `description` is automatically generated in the yaml file, when I see it is not used anymore

**Comment by lucasrodes** [comment]
Hey! Should I just reopen for a small issue then?

- [ ] The create playground notebook option in wizard doesn't generate a Jupyter Notebook

(copied from https://github.com/owid/etl/issues/1566#issuecomment-1723433161)

**Comment by spoonerf** [comment]
It's not super clear to me whether I should be hitting return after filling in each section or not? It seems like a good thing to somewhat validate each entry, but also the form completed unexpectedly early when pressing it, meaning I skipped the last couple of sections.

**Comment by paarriagadap** [comment]
- [ ] The video explaining the harmonization tool is outdated:

> Harmonize country names with the following command (assuming country field is called country). Check out a [short demo](https://drive.google.com/file/d/1tBFMkgOgy4MmB7E7NmfMlfa4noWaiG3t/view) of the tool

**Comment by paarriagadap** [comment]
- [ ] I think the options to add regions and population data available on walkthrough should also be available on wizard as well.

**Comment by lucasrodes** [comment]
@spoonerf, good point, but unfortunately, clicking on ENTER is equivalent to submitting the entire form. It will be submitted if the form is valid (i.e., all required fields are present).

**Comment by lucasrodes** [comment]
@pabloarosado 

>  I think the options to add regions and population data available on walkthrough should also be available on wizard as well.

Which options do you refer to here? I just executed `walkthrough` and there aren't - at least on my end - any options to add regions or population datasets.

**Comment by paarriagadap** [comment]
@lucasrodes Maybe it was removed recently, but they were available as check boxes at the # end of walkthrough garden

**Comment by lucasrodes** [comment]
- [ ] Improve license selector: remove spurious options, add missing ones
- [ ] Have namespace as a selector, like the license. Should be based on the names of the folders in `etl/steps/data/channel/`

**Comment by paarriagadap** [comment]
- [ ] In `etl-wizard snapshot`, in the License section, there is the `©producer year` option, but it prints `©producer year` and not `©{producer} {year}` like `origin.attribution`

**Comment by paarriagadap** [comment]
- [ ] It would be nice to have the `Select local file to import option` with the option to navigate the OS and select the file instead of writing a path that can be long
<img width="666" alt="image" src="https://github.com/owid/etl/assets/63430031/6273917e-f31f-4351-ad2f-53fe977a7e0a">

**Comment by paarriagadap** [comment]
- [ ] In the chart revisions baker, I tried to move one chart from [here](https://owid.cloud/admin/datasets/1070) to [here](https://owid.cloud/admin/datasets/6096) and after the balloons 🎈 : the links for live and staging (I don't use local) were not updated with the new chart in the chart revision tool.
- [ ] I restarted and tried again: I got this error message: 
```
Something went wrong! (MySQLdb.IntegrityError) (1062, "Duplicate entry '1904-3-3-1' for key 'suggested_chart_revisions.chartId'") [SQL: INSERT INTO suggested_chart_revisions (chartId, createdBy, originalConfig, suggestedConfig, status, createdAt, updatedAt) VALUES (%s, %s, %s, %s, %s, %s, %s)] [parameters: (1904, 59, '{"id": 1904, "map": {"colorScale": {"baseColorScheme": "YlGn", "binningStrategy": "manual", "customNumericColors": [null, null, null, null, null, nul ... (2786 characters truncated) ... 71}], "sourceDesc": "The World Bank", "isPublished": true, "selectedEntityNames": ["Azerbaijan", "Antigua and Barbuda", "Rwanda", "Sudan", "Brazil"]}', '{"id": 1904, "map": {"colorScale": {"baseColorScheme": "YlGn", "binningStrategy": "manual", "customNumericColors": [null, null, null, null, null, nul ... (651 characters truncated) ... 47}], "sourceDesc": "The World Bank", "isPublished": true, "selectedEntityNames": ["Azerbaijan", "Antigua and Barbuda", "Rwanda", "Sudan", "Brazil"]}', 'pending', datetime.datetime(2023, 11, 1, 16, 19, 38, 280631), datetime.datetime(2023, 11, 1, 16, 19, 38, 280631))] (Background on this error at: https://sqlalche.me/e/14/gkpj)
```

I saved [this draft chart](https://owid.cloud/admin/charts/7355/edit) to replicate ([indicator here](https://owid.cloud/admin/variables/45071)).

**Comment by lucasrodes** [comment]
- [ ] Enable setting update frequency to zero (= this dataset won't be updated anymore) [[slack thread]](https://owid.slack.com/archives/C05KQ0LE05V/p1699279627642059)

**Comment by paarriagadap** [comment]
- [ ] The links "Explore dataset" in `etl-wizard charts` don't actually work. They have the format `http://localhost:8053/None/datasets/{dataset_id}/`

<img width="1572" alt="image" src="https://github.com/owid/etl/assets/63430031/06fdc366-f09a-486a-8ca2-37f6d80c698a">

**Comment by paarriagadap** [comment]
- [ ] I can't find one indicator in `etl-wizard charts`: when trying to migrate from `[818136] $30 a day - Number not in poverty (Estimated)` to `[818156] $30 a day - Number not in poverty (Smoothed)` I actually can't find the latter. Dataset is [[6341]](https://owid.cloud/admin/datasets/6341)

**Comment by paarriagadap** [comment]
Probably there's not much to do, but

- [ ] Variables from datasets 489 and 6398 can't be compared in the Charts upgrader because of this error:
<img width="1794" alt="image" src="https://github.com/owid/etl/assets/63430031/2e55eec5-aaaf-41c4-9be1-db0a4c72e5e6">

I see that the csv [here](https://owid.cloud/admin/datasets/489) includes `..` for some data points.

**Comment by lucasrodes** [comment]
@paarriagadap, thanks for reporting. This issue is, as you mention, because there are non-numeric values in the dataset. Currently, the explore mode only supports comparing numerical values. We could compare categorical ones in the future if we needed to.

However, in this case, I think the indicators should be numbers and `..` should be `NaN` instead. That way, the indicator would be recognised as numeric. Also, why do we have `..` there? I bet that's a legacy dataset, and we won't be fixing this (removing `..`). Unfortunately, implementing some logic to replace `..` so that we can compare the data is too specific and don't think we should do it.

**Comment by paarriagadap** [comment]
- [ ] The relative difference column in the explore variable mappings option in the Charts upgrader is not colored anymore. I thought it was a nice addition to draw immediate attention to wildly different numbers.
<img width="940" alt="image" src="https://github.com/owid/etl/assets/63430031/56af4ebd-818b-4157-bae7-5308d14f9983">

**Comment by lucasrodes** [comment]
Chart revision:

- [ ] Enable some option to "load from mapping", where the user can load a json with the variable mapping
- [ ] If no OPENAI_API_KEY is available, the app will crash miserably if trying to get chatGPT suggestions. Instead, it should just show a warning and let the user continue!

**Comment by pabloarosado** [comment]
I see that some classic issues with the chart revision tool still happen occasionally:
* The sorting in bar charts is not respected (e.g. [this chart](http://staging-site-update-faostat-data/admin/charts/4926/edit)):
![Screenshot 2024-03-22 at 09 27 16](https://github.com/owid/etl/assets/12246978/460f40bf-844e-4522-b484-4132bc2ce6fa)
* The min and max values in the timeline are not respected (e.g. [this chart](http://staging-site-update-faostat-data/admin/charts/1559/edit)):
![Screenshot 2024-03-22 at 09 52 17](https://github.com/owid/etl/assets/12246978/093253ce-b8e6-4941-92ad-0178ce3a1632)

It's not urgent, and I have already manually fixed the affected cases. I just mentioned them here so we are aware that these minor things still happen.

**Comment by Marigold** [comment]
I'll be looking at chart revisions this week, so I will take a look if there's an easy fix for that. (Examples like that are super useful by the way)

**Comment by lucasrodes** [comment]
[Issue raised by pablo](https://github.com/owid/etl/issues/1563#issuecomment-2015462890) might be related to this one: https://github.com/owid/etl/issues/867 (meant to post this here)

cc. @Marigold

**Comment by paarriagadap** [comment]
Is the sorting in bar charts in the approval tool [fixed here](https://github.com/owid/etl/pull/964)? Because I still find the issue:
<img width="1441" alt="image" src="https://github.com/owid/etl/assets/63430031/a5da00fd-2230-42eb-955d-d8cb8162ffc8">

**Comment by paarriagadap** [comment]
- [ ] When using wizard snapshots, if I save the title with `:` it generates an error in the yaml file. It can be solved if the content is rendered in quotes.

<img width="918" alt="image" src="https://github.com/owid/etl/assets/63430031/5937e049-3f1e-4dbb-b59c-9165452f7453">
<img width="922" alt="image" src="https://github.com/owid/etl/assets/63430031/bcc572fa-9177-4c81-82a1-6749b6f97c5f">

**Comment by paarriagadap** [comment]
- [ ] Minor: The entity harmonizer might look a bit confusing when you add a custom name because the suggested name is still shown. Would it be possible to remove it when filling the custom box?
<img width="665" alt="image" src="https://github.com/owid/etl/assets/63430031/1d11b2e9-8072-4895-a799-a93b67082ad4">

**Comment by paarriagadap** [comment]
- [ ] Suggestions in the entity harmonizer suggest wrong options when there is a parenthesis after a recognizable country
<img width="620" alt="image" src="https://github.com/owid/etl/assets/63430031/7471ea17-0b0f-4f95-8c87-3022ca227e55">

**Comment by lucasrodes** [comment]
>  When using wizard snapshots, if I save the title with : it generates an error in the yaml file. It can be solved if the content is rendered in quotes.

Thanks for reporting @paarriagadap. I think this by design, otherwise the linter gets confused with the keyword. As you say, you can solve this by putting the text in quotes (`"`) or using a multiline string (`|-`, `>`, etc.)

**Comment by paarriagadap** [comment]
- [ ] [Wizard expert](https://etl.owid.io/wizard/Expert) is failing with this error:
```
ValueError: Property variable.display.numDecimalPlaces has no type!
Traceback:
File "/home/owid/etl/.venv/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 600, in _run_script
    exec(code, module.__dict__)
File "/home/owid/etl/apps/wizard/pages/expert/app.py", line 13, in <module>
    from apps.wizard.pages.expert.prompts import (
File "/home/owid/etl/apps/wizard/pages/expert/prompts.py", line 52, in <module>
    {render_indicator()}
File "/home/owid/etl/etl/docs.py", line 161, in render_indicator
    documentation = render_props_recursive(
File "/home/owid/etl/etl/docs.py", line 115, in render_props_recursive
    text += render_props_recursive(
File "/home/owid/etl/etl/docs.py", line 115, in render_props_recursive
    text += render_props_recursive(
File "/home/owid/etl/etl/docs.py", line 123, in render_props_recursive
    text += render_prop_doc(prop, prop_name=prop_name, level=level)
File "/home/owid/etl/etl/docs.py", line 65, in render_prop_doc
    raise ValueError(f"Property {prop_name} has no type!")
```

**Comment by paarriagadap** [comment]
- [ ] (Minor) After running the Snapshot step via streamlit, it could have the button to continue in Meadow
<img width="459" alt="image" src="https://github.com/owid/etl/assets/63430031/3f05ab69-0abd-4ba5-b379-b2325d07680f">

**Comment by paarriagadap** [comment]
Indicator Upgrader didn't recognize the version of a dataset updated this year: [PR](https://github.com/owid/etl/pull/3569)
![Image](https://github.com/user-attachments/assets/85eb345c-ef0f-4365-b375-af6126ce9e79)

**Comment by lucasrodes** [comment]
@pabloarosado in case you want to look into the detection algorithm

**Comment by pabloarosado** [comment]
Hi @paarriagadap thanks for reporting this, I've fixed it in [your branch](https://github.com/owid/etl/pull/3569). There were two issues:
* The archive dag file `dag/archive/poverty_inequality.yml` was created, but it wasn't added to `dag/archive/main.yml`, and therefore it was ignored.
* Then I also realised that there were two steps that were both in the active and in the archive dag, namely:
  * `data://explorers/lis/latest/luxembourg_income_study`
  * `data://explorers/wid/latest/world_inequality_database`
  If you think this was caused by wizard (which would be a bug), let me know. Thanks!

**Comment by paarriagadap** [comment]
Hi @pabloarosado. Ah, thanks for the fix. I should have forgotten about adding the dag file in the archive main.yml

Regarding those `latest` steps, yes, I think that when I use the tool to update steps, these latest steps are moved to the archive and disappear from the "live" yaml. At least I have seen that I need to re-add those steps to the latter.

**Comment by pabloarosado** [comment]
Small bug in chart animation: Sometimes the first frame of a line chart animation becomes a bar chart (even though "Show single year" is deactivated).
![Image](https://github.com/user-attachments/assets/a7f279c5-14c5-4f1b-a580-137ec1cab0a3)

**Comment by paarriagadap** [comment]
In Wizard Express we could have the _Generate playground notebook_ option in case we want it. I use to create them in the separate steps.

**Comment by lucasrodes** [comment]
>In Wizard Express we could have the Generate playground notebook option in case we want it. I use to create them in the separate steps.

I can add it back.

Some follow-up questions:
- Do you use it only in Garden?
- Do you use the interactive shell in VS Code? If so, where does it fall short compared to the playground notebook?

**Comment by paarriagadap** [comment]
Hi @lucasrodes!

- No, I also can use it in meadow, to check the resulting table and table structure
- No, I don't use it much

**Comment by Marigold** [comment]
> Do you use the interactive shell in VS Code? If so, where does it fall short compared to the playground notebook?

Sorry to chime in! I also use `playground.ipynb` notebooks a lot (although for different use cases I guess). The main advantage over interactive shell is that it keeps your "history" in cells. If I switch to a different task and then go back, it's harder to get to the same state.

**Comment by pabloarosado** [comment]
> > Do you use the interactive shell in VS Code? If so, where does it fall short compared to the playground notebook?
> 
> Sorry to chime in! I also use `playground.ipynb` notebooks a lot (although for different use cases I guess). The main advantage over interactive shell is that it keeps your "history" in cells. If I switch to a different task and then go back, it's harder to get to the same state.

Sorry to chime in as well! :-D
We could actually also achieve the same thing with `playground.py` files, ignored in `.gitignore`. The only difference would be that you would not be able to see the output cells. But, unless you are working on very big datasets that take a long time to run, you can simply re-run any playground code to produce the same outputs again.
That said, I have nothing against keeping the option of `playground.ipynb`, if others find it useful. But I would prefer to keep all `ipynb` files out of the repos, and stick to `.py` for simplicity.

**Comment by Marigold** [comment]
Yep, that's a good idea. I'll give it a try next time.

> But I would prefer to keep all ipynb files out of the repos, and stick to .py for simplicity.

All `playground.*` files are in .gitignore anyway, so we're not going to commit anything.

**Comment by pabloarosado** [comment]
* [ ] Improvement on Producer analytics: [Ed pointed out](https://owid.slack.com/archives/C014SGT6W9X/p1734355821172419?thread_ts=1734090626.333919&cid=C014SGT6W9X) that it would be good to let the user see analytics not only at the data producer level, but also at the data producer & dataset level.
One option would be to have a toggle to show either data producer or data producer & data product in the main table.

**Comment by paarriagadap** [comment]
Indicator Upgrader doesn't show the version of some older datasets (https://github.com/owid/etl/pull/3843):

<img width="1492" alt="Image" src="https://github.com/user-attachments/assets/d2692d50-d013-4c8d-90b7-3f9bfc0d5944" />

**Comment by paarriagadap** [comment]
It would be cool to have in Chart-Diff a field to filter by variable name / ID, especially for massive updates such as WDI. I want to take a look only at the changes generated by the GDP/GDP per capita variable

<img width="1785" alt="Image" src="https://github.com/user-attachments/assets/5688aea8-c8ac-4e65-b18f-2c2511df2e60" />

**Comment by Marigold** [comment]
> It would be cool to have in Chart-Diff a field to filter by variable name / ID, especially for massive updates such as WDI. I want to take a look only at the changes generated by the GDP/GDP per capita variable

I'll add it, I could use that filter too.

**Comment by paarriagadap** [comment]
Chart-diff currently shows some charts in the wrong format. See http://staging-site-update-wid-nov-2024/etl/wizard/chart-diff?chart_id=5996. The chart in [live admin](https://admin.owid.io/admin/charts/5996/edit) doesn't have the problem, but the one in [staging-master](http://staging-site-master/admin/charts/5996/edit) does

<img width="866" alt="Image" src="https://github.com/user-attachments/assets/ac5e30a9-9c8f-4728-a5f0-976c162f2d25" />

**Comment by Marigold** [comment]
Thanks for reporting. I have no idea what could have happened. It almost looks like we ran into hash collision, but that would be extremely unlikely. I fixed it on `staging-site-master`. If you run into such case again, let me know please 🙏

**Comment by paarriagadap** [comment]
It seems that the data step tools is duplicating the ds_garden.save() code at the end of meadow or garden steps. It has happened twice today, I will report if I see where it happens exactly if it happens again:

<img width="328" alt="Image" src="https://github.com/user-attachments/assets/1a7f6d9f-a5e0-4bbd-8493-f96f0d3a409d" />

**Comment by Marigold** [comment]
> Anomalist doesn't work when I want to compare versions with a pre-ETL dataset

Rather than adding support for pre-ETL datasets, we could migrate it to ETL. Ping me if this is important enough and I'll take a look.

**Comment by paarriagadap** [comment]
Thanks @Marigold. This example is not that important to have a migration, but I will ping if I need it in the future.

**Comment by paarriagadap** [comment]
Anomalist is confused understanding which source is the old version and which one is the new one.

In this specific case, I decided to update the version of the data https://github.com/owid/etl/pull/3569, because the data had errors that took months to fix. So when I changed the version I used Indicator Upgrader to revert to the old version and then back to the newest version.

With that, Anomalist, doesn't quite know which one is the old and which one is the new and actually it shows the three datasets by default:
http://staging-site-update-wid-nov-2024/etl/wizard/anomalist

<img width="1441" alt="Image" src="https://github.com/user-attachments/assets/a0d85f41-28a3-4529-9385-8d2e8b23e15a" />

**Comment by pabloarosado** [comment]
Explorer diff seems to be doing something odd when there is a mismatch between the views of the old and new explorer.

In [this case](http://staging-site-data-agriculture-update/etl/wizard/explorer-diff?explorer=global-food&global-food_Food=Buffalo+hides&global-food_Metric=Yield&global-food_Per+Capita=false&global-food_Unit=Fat+per+day), I loaded the global-food explorer, and [the old one](https://ourworldindata.org/explorers/global-food?Food=Buffalo+hides&Metric=Yield&Per+Capita=false&country=USA~DEU~FRA~GBR~BRA~ZAF) had data for "Buffalo hides" "Yield". [The new explorer](http://staging-site-data-agriculture-update/admin/explorers/preview/global-food?Food=Buffalo+hides), however, doesn't have that option (there's no data for that combination in the latest update). But, instead of showing nothing, explorer diff is showing some other data (it shows "Production" which is the closest available option in that dropdown).
I suppose this is what the warning is referring to ("⚠️ This specific combination of options does not exist in the explorer views. The explorer may show unexpected results."), but the current implementation can be confusing.

![Image](https://github.com/user-attachments/assets/f848bb66-83b5-4a09-8154-d44fa784cd3f)

**Comment by Marigold** [comment]
@pabloarosado that's a known issue, unfortunately. It was technically challenging, but I'll take another look at it with a fresh perspective.

**Comment by Marigold** [comment]
The first prototype of explorer diff from https://github.com/owid/etl/pull/4108 is live. Below is a list of missing features we might implement in the future

## Missing Features  
- [ ] If a selection is missing from "views," explorers shouldn't render rather than displaying an arbitrary selection.  
- [ ] Show top views based on pageviews.  
- [ ] "Random view" should be sampled proportionally to pageviews.  
- [ ] Missing country selection (changing a selection always shows default countries)

**Comment by Marigold** [comment]
By @pabloarosado after a faostat indicator upgrader [nightmare](https://owid.slack.com/archives/C46U9LXRR/p1742548873984849).

> One possible idea to avoid potential issues like this would be that Indicator Upgrader by default hides mappings that match 100%, and highlights those that do not match 100%

We should also add an assertion for checking if the final mapping is 1:1. If not, we should debug whether it was human error or a bug.

**Comment by pabloarosado** [comment]
Thanks @Marigold, I think it's in principle possible to have a mapping that is not 1:1 (e.g. two indicators now have become the same one in the latest release). So an assertion may not be the right solution, but certainly raising a warning to pinpoint these unusual cases would help catch these issues (that can become a nightmare if uncaught at the right time).
However, I think that mappings 1:n might cause issues in the grapher chart admin, similar to what I experienced here.

https://github.com/user-attachments/assets/976decdf-034b-41ea-a48c-e94c0398f572
https://github.com/user-attachments/assets/3fb4eb5c-fa9a-4964-90c5-5ef94cf04284

**Comment by pabloarosado** [comment]
It would be good if the chart animation tool could handle drafts. Fiona requested it, and I agree that it would be useful, since you may want to create a DI based on an unpublished chart.

**Comment by paarriagadap** [comment]
- [ ] Sadly, the TSV diff in Explorer diff is not useful when there are many changes involved. It is very hard to see where the changes are. I understand this is a Streamlit limitation, but I report it in case there is something we can do. [Example](http://staging-site-data-world-bank-pip/etl/wizard/explorer-diff?explorer=poverty-wb)

<img width="1670" alt="Image" src="https://github.com/user-attachments/assets/862664c3-11cf-4b27-9301-9e2cb154c705" />

**Comment by Marigold** [comment]
Hmm, I'll see if showing YAML-like diff instead of TSV could show more useful output.

**Comment by Marigold** [comment]
@paarriagadap I've added `YAML Diff` tab in addition to `TSV Diff`. It should be more meaningful for some explorers, give it a try!

**Comment by paarriagadap** [comment]
Chart Diff is not showing if a chart is a draft. In this case, for example http://staging-site-data-global-military-spendin/etl/wizard/chart-diff?chart_id=2567

**Comment by paarriagadap** [comment]
In [Indicator Upgrader](http://staging-site-data-world-bank-pip/etl/wizard/indicator-upgrade), I have a long list of indicators to migrate, with a different name structure. Because of that, decided to migrate some to come back later. The first batch migrated well, but when I return to the tool I still have the full list of indicators to migrate, and it should be a smaller set of them.

See for example the indicator World Bank PIP region, alredy migrated from [here](http://staging-site-data-world-bank-pip/admin/variables/990543) to [here](http://staging-site-data-world-bank-pip/admin/variables/1036163). When I re-run Indicator Upgrader I can still see it on the list. The same is happening to all the indicators I already migrated.

<img width="1408" alt="Image" src="https://github.com/user-attachments/assets/97bbf46f-a9f8-4281-97b5-8b4a5059ec79" />

**Comment by paarriagadap** [comment]
Changes are not saved in charts when I press "Refresh charts" in Chart-diff. In fact, the changes are reversed. See video:

https://github.com/user-attachments/assets/7e2ceb1e-5139-4552-b152-1096c46c9c0c

**Comment by paarriagadap** [comment]
The one below I am posting it in [tech-issues](https://owid.slack.com/archives/C46U9LXRR/p1747638193664529), given that it's most likely a Grapher issue

**Comment by paarriagadap** [comment]
Inheritance configuration is not synced with master after setting it in staging. I realized [this PR](https://github.com/owid/etl/pull/4245) didn't carried inheritance to master and I would really like it to work in this one: https://github.com/owid/etl/pull/4496.

**Comment by Marigold** [comment]
> Inheritance configuration is not synced with master after setting it in staging

Sorry for my ignorance, but what is _inheritance configuration_? Ideally with an example?

**Comment by paarriagadap** [comment]
Ah, no worries. Inheritance is default Grapher configurations coming straight from metadata. Sophia explains [here](https://owid.slack.com/archives/C014SGT6W9X/p1725539193555019?thread_ts=1725539193.555019&cid=C014SGT6W9X).

So for example I set inheritance for [this chart](https://admin.owid.io/admin/charts/5111/edit) (and many others) to use the default title, subtitle, note and all the FAUST from the metadata. These fields are then linked and not manually modified, but chart sync didn't pick that up and the fields are "unlinked" again

**Comment by pabloarosado** [comment]
The "Config diff" and the "Metadata differences" buttons in chart diff are not working (e.g. in [this server](http://staging-site-data-income-group-grapher/etl/wizard/chart-diff?chart-diff-sort-by-results=relevance&show_reviewed=&show_all=), but I've experienced that in different staging servers).

![Image](https://github.com/user-attachments/assets/f75761d4-ec4c-4adb-9210-555aefa10f25)

**Comment by paarriagadap** [comment]
Indicator Upgrader forgets the indicator that will replace the old one when I navigate between pages. See this example:

https://github.com/user-attachments/assets/4606bb6b-c267-4911-a721-c1455d046b2c

**Comment by paarriagadap** [comment]
Inheritance is not respected when I migrate to a new indicator and this indicator does not have the same metadata as the old one. For example, in [this chart](http://staging-site-data-world-bank-pip-f7d786/admin/charts/5181/edit) I set inheritance and linked the title, subtitle, note, entities, brackets, etc, but after migrating to a new indicator in different prices it unlinks subtitle, note and map brackets. See the video:

https://github.com/user-attachments/assets/4e9c295b-cf6e-4444-b713-f6af5a813115

**Comment by paarriagadap** [comment]
In [this PR](https://github.com/owid/etl/pull/4496) I have found myself approving charts in chart-diff more that I would need to. See this example: my last revision was 5 days ago, it was approved and I see it is pending again for changes supposedly done yesterday. 
<img width="537" alt="Image" src="https://github.com/user-attachments/assets/c8eda547-e86c-4865-ba38-77a82a3f065e" />

<img width="1768" alt="Image" src="https://github.com/user-attachments/assets/000ac5af-e4c6-4583-8a76-529d3a902b30" />

**Comment by Marigold** [comment]
@paarriagadap it could be due to the [migration in owid-grapher](https://github.com/owid/owid-grapher/pull/5006) that changed configs (cc @marcelgerber). (I haven't double-checked that)

**Comment by paarriagadap** [comment]
@Marigold You're right, that could be it. I needed to change the brackets in the ETL with Marcel's change

**Comment by paarriagadap** [comment]
Links for specific chart-view comparisons do not work after approving the new versions, probably because they are now hidden after the green check.

See this video for [this example](http://staging-site-data-wb-poverty-projections/etl/wizard/chart-diff)

https://github.com/user-attachments/assets/ae8c5236-62b7-4850-9975-516338910ec2

**Comment by paarriagadap** [comment]
(Let me know if you now prefer to have separate issues and close this one)
